### ISSUE TITLE
[Placeholder] Hardened files operation handling (moving, copying, uploading?)

### ISSUE THREAD

**Issue by marcoambrosini** [nextcloud/server#41790]
cc @nextcloud/designers

**Comment by jancborchardt** [comment]
Sounds good as talked about :) @marcoambrosini do you want to pick this up?

**Comment by marcoambrosini** [comment]
@nextcloud/designers could you please take a look at the reasoning in the collective and give me some feedback?

**Comment by juliushaertl** [comment]
> Anything else you think it's worth including? @juliushaertl mentioned account migrations, I think.

Ownership transfer of files

**Comment by szaimen** [comment]
@marcoambrosini looks already really nice, great mockup! :)
However, I would probably add a H2 heading like Task manager to the top of the popup and also do the chips fullfil our requirements in terms of clickable area for mobile?

**Comment by nimishavijay** [comment]
Super cool idea! :) Between the 2 approaches, I prefer approach 1 (this one) as it helps with people switching between apps when a task is in progress. I fear that it might give an impression of the entire instance being a bit "slow" as the idea is that "everything is scheduled". What do you think? 

Some more high-level questions:
- [ ] Would the icon be shown all the time or only when there are tasks in progress? 
- [ ] What would "Open details" do?
- [ ] Is the scope of this limited to just viewing the tasks or can they be paused/restarted/cancelled etc?
- [ ] I'm wondering if it makes sense for this to be combined with the notifications, for example as maybe another tab (just an idea, to help with keeping the top right corner features simple)

What do you think? :)

**Comment by nickvergessen** [comment]
> Talk recording server processes

I don't think this is a good fit. It takes below seconds to make the recording available.

Transcriptions are another story, but that is "AI processes", so don't see a big usecase for Talk at the moment.

**Comment by jancborchardt** [comment]
> > I'm wondering if it makes sense for this to be combined with the notifications, for example as maybe another tab (just an idea, to help with keeping the top right corner features simple)
> 
> While I get the point and this is already how the desktop client does it, I beg you to not combine this. The concepts are totally different, APIs are totally different, Notifications is even an app that can be disabled, ...
Also it makes it much less visible that e.g. a task was started, etc.


@nimishavijay @nickvergessen just talked with @karlitschek about the proposal and we also came to the conclusion it would be best to combine it with notifications, rather than introducing yet another item to the top bar.

This is similar e.g. on Android, where for example uploads and downloads are notifications while in progress, and then also when done.
Adding yet another item to the top bar would really be a bit much. We need to take care to not make the interface too busy.

**Comment by nickvergessen** [comment]
> just talked with karlitschek about the proposal and we also came to the conclusion it would be best to combine it with notifications, rather than introducing yet another item to the top bar.
> 
> This is similar e.g. on Android, where for example uploads and downloads are notifications while in progress, and then also when done.
> Adding yet another item to the top bar would really be a bit much. We need to take care to not make the interface too busy.

I totally get that. just saying that something that is rather stable and easy maintainable at the moment, will turn into a beast that operates on 2 different data stores and data structures. I'm currently maintaining the notifications app in it's current form, but I can not be of any help with a change like this.

We can of course separate the "header menu" from the notifications logic and then have both features, task manager and the notifications hook into it and doing their things. But API-wise (with pushes, mobile support, desktop clients, etc.) the concepts should be separated and not mixed into each others.

**Comment by jancborchardt** [comment]
Just for the record: This is not something planned right now, so moved it out of "Design phase".

---

> we could make space for this feature in the header by removing the contacts search, which has been duplicated functionality for a long time right next to the global search.

@marcoambrosini the People menu is not just "Contacts search". It aims to connect you to colleagues and other people on the instance that you work with a lot. We should improve it rather than remove it.

> Edit: frankly, it's not just a matter of technical difficulty. From a UX perspective, I think that task manager and notifications are complex enough to have their own element too.

Android and iOS have progress like this as notifications. We don’t need to make it more complicated.

---

> I totally get that. just saying that something that is rather stable and easy maintainable at the moment, will turn into a beast that operates on 2 different data stores and data structures.

@nickvergessen yup, understood. How it is done technically is one thing, what would be nice if it’s displayed in the interface in the same place.

**Comment by marcoambrosini** [comment]
> Android and iOS 

Mobile notifications is a bit of an arbitrary as a reference, IMO. No collaboration suite mixes these "tasks" with notifications because they are both complicated and unrelated (see the collectives page linked above for the references). 

> People menu is not just "Contacts search"

<img width="350" alt="Screenshot 2024-01-17 at 19 33 58" src="https://github.com/nextcloud/server/assets/26852655/56282fa2-28c7-48c7-b92c-eb599860e51c">

That is what it says, though. And it's right there beside the search icon. 
It's also very likely that most new people will click on the search icon to find and interact with someone. 

> what would be nice if it’s displayed in the interface in the same place

Not saying it can't be done, but it seems that the argument in favor (it's too crowded up there) is pretty weak, especially when we're allocating that space for duplicated functionality.

**Comment by drascom** [comment]
thumbs up for idea it is very neccessary ui feature. synology has something like that for ages. <img width="1001" alt="image" src="https://github.com/user-attachments/assets/0878e815-b13f-4405-89b8-685f54f53359">

**Comment by marcoambrosini** [comment]
I brainstormed this a bit more with @jancborchardt and we think it would be a good idea to merge this with notifications.

**Comment by nickvergessen** [comment]
While it might be "a good idea" design/UX-wise (not saying yes/no to that), that is a definite **No** from API side for me. Interoperability with clients, how push notifications work and the data structure to aligning with OS and WebBrowser notifications, as well as scalability are nothing we can sacrifice.

That being said, all OS I ever used (Ubuntu, Mint, Windows, Mac) have "progress" in different sections and not molded into notifications.

**Comment by MrRinkana** [comment]
I think a separate icon that only appears when something is being done is the best looking, but integrated in notifications is also nice, as long as the icon indicates a action is being performed at a glance (not just a notification dot).

**Comment by marcoambrosini** [comment]
@MrRinkana we need to integrate it with notifications mostly for real estate reasons. Otherwise I'd also preferred having a dedicated element for these

**Comment by ThaDaVos** [comment]
When can we expect this? Currently having no way to check how far along my move of a 280GB folder is - other than refreshing and hoping it's done already - I can't even see the folder in the new location yet (probably gets updated as soon as it's done)

**Comment by skjnldsv** [comment]
@ThaDaVos no milestone, and labels still says `needs triage`.
There is no ETA

**Comment by surapuramakhil** [comment]
@marcoambrosini can you share status of this? like if any PR has been raised? is this planned? or any ETA when we can see this live etc.

**Comment by marcoambrosini** [comment]
@surapuramakhil it's not planned for now. You can check the status in the projects field in the right sidebar

**Comment by susnux** [comment]
From #46078

> Hello,
> i'm using nextcloud from about version 20. Still missing good looking file upload via web browser _(like on mega.nz)_:
> 
> ![image](https://github.com/nextcloud/server/assets/43697578/ad60aaba-6fdf-4d3b-b2b5-cbae281b6438)
> 
> ### Features:
> 1. Visually atractive
> 2. See list files uploading
> 3. See the speed of uploading
> 4. See time to end of upload
> 5. See the total percentage of uploading
> 6. A possibility tu pause or resume whole upload or only selected files
> 7. Delete some files from list
> 8. Dynamicly in floating window upload progress from anywhere on the web


As this is essential what this issue is about lets merge it here.

**Comment by sptcguy** [comment]
Have any thoughts been given to how this could be integrated with a queue/task runner? For some of the more computationally expensive jobs it'd be great if we could farm them out to a job server (like gearman).  I realize ultimately this would be a different issue/discussion.

**Comment by popos123** [comment]
Thank You @susnux for attaching my post from last year to show the problem more clear.
So this problem is konwn from couple years form now, **is there any progress?**
